src/classes/OpportunityTriggerHelper.cls

```diff
@@ -0,0 +1,421 @@
+                                    message1 = sensorType + ' is moved from Shipment Account ' + opp.Shipment_Account__c + ' to Installation Account ' + opp.AccountId;                                    
+                                }
+                                else {
+                                    message1 = sensorType + ' is moved from Installation Account ' + opp.AccountId + ' to Shipment Account ' + opp.Shipment_Account__c;                                   
+                                }
+                            }
+                            else {
+                                for (Database.Error error : result.getErrors()) {
+                                    message2 = error.getStatusCode() + ': ' + error.getMessage();
+                                }
+                            }
+                        }
+                    }
+                    else if (sensorName.substring(1,3).isNumeric()) {
+                        sensorType = 'Automobile Tag';
+                        if (flag == true) {
+                            if (result.isSuccess()) {
+                                message1 = sensorType + ' Inserted Under Installation Account : ' + (String)recordName.get('Account__c');
+                            }
+                            else {
+                                for (Database.Error error : result.getErrors()) {
+                                    message2 = error.getStatusCode() + ': ' + error.getMessage();
+                                }
+                            }
+                        }
+                        else {
+                            if (result.isSuccess()) {
+                                if (opp.AccountId == (String)recordName.get('Account__c')) {
+                                    message1 = sensorType + ' is moved from Shipment Account ' + opp.Shipment_Account__c + ' to Installation Account ' + opp.AccountId;                                    
+                                }
+                                else {
+                                    message1 = sensorType + ' is moved from Installation Account ' + opp.AccountId + ' to Shipment Account ' + opp.Shipment_Account__c;                                   
+                                }
+                            }
+                            else {
+                                for (Database.Error error : result.getErrors()) {
+                                    message2 = error.getStatusCode() + ': ' + error.getMessage();
+                                }
+                            }
+                        }
+                    }
+                }
+
+                // Add message to HTML Body
+                htmlBody1 += '<tr><td>' + sensorType + '</td><td>' + sensorName + '</td><td>' + message1 + '</td></tr>';                        
+                htmlBody2 += '<tr><td>' + sensorType + '</td><td>' + sensorName + '</td><td>' + message2 + '</td></tr>';   
```

Comment:
This is the reason you are getting multiplr rows for the records in the email. Why are you using 2 html body with same value.

Reply:
Use "if" to check which message to use here .